core/display.js

```diff
@@ -741,6 +742,16 @@
             this._drawCtx.putImageData(img, x - vx, y - vy);
         },
 
+        _renderQ_push: function (action) {
+            this._renderQ.push(action);
```

Comment:
I wonder if it would be more efficient here to process immediately instead of pushing and then popping if the render queue is empty?  That might take a bit of rewriting.  Just a thought for later.